app/src/main/java/com/techyourchance/coroutines/exercises/homework/ui/CoroutineConfigExcerciseFragment.kt

```diff
@@ -0,0 +1,63 @@
+package com.techyourchance.coroutines.exercises.homework.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ProgressBar
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.techyourchance.coroutines.R
+import com.techyourchance.coroutines.common.BaseFragment
+import com.techyourchance.coroutines.exercises.homework.useCase.EmulateNetworkCallUseCase
+import kotlinx.coroutines.*
+
+class CoroutineConfigExcerciseFragment : BaseFragment() {
+
+    private lateinit var emulateNetworkCallUseCase: EmulateNetworkCallUseCase
+    private lateinit var recyclerExercise: RecyclerView
+    private lateinit var progressBar: ProgressBar
+    private var resultList: ArrayList<String> = ArrayList()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        emulateNetworkCallUseCase = compositionRoot.emulateNetworkCallUseCase
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+
+        val view = inflater.inflate(R.layout.fragment_coroutine_exercise, container, false)
+        view.apply {
+            recyclerExercise = findViewById(R.id.exercise_recycler_view)
+            progressBar = findViewById(R.id.progress_bar)
+        }
+
+        with(recyclerExercise){
+            layoutManager = LinearLayoutManager(context)
+            adapter = ExerciseAdapter(resultList)
+        }
+        initValues()
+
+        return view
+    }
+
+    private fun initValues(){
+        viewLifecycleOwner.lifecycleScope.launch(Dispatchers.Main.immediate) {
+            withContext(CoroutineName("init coroutine")){
+                resultList.addAll(emulateNetworkCallUseCase.emulateNetworkCall())
```

Comment:
So, in which context are you making this "network call"?

Reply:
I see now you are switching the context on the UseCase, to `Dispatchers.IO`. Now I'm curious about two other things:
 1 - It is not needed to set `Dispatchers.Main.immediate` since it is defaulted for `lifecycleScope`
 2 - Why using another coroutine on the line `48`? seems to me that you can achieve the same result without the coroutine named "init coroutine"